src/main/java/io/appform/dropwizard/actors/actor/MessageMetadata.java

```diff
@@ -1,13 +1,35 @@
 package io.appform.dropwizard.actors.actor;
 
+import java.util.Date;
+import java.util.Map;
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
 @Data
+@Builder
+@NoArgsConstructor
 @AllArgsConstructor
 public final class MessageMetadata {
 
+    private String contentType;
+    private String contentEncoding;
+    private Map<String,Object> headers;
+    private Integer deliveryMode;
```

Comment:
use primitives wherever possible

Reply:
With primitives, the values will be set to 0 by default. And, clients won't be able to figure out if it is set as 0 or left unset. With boxed types, the null check can help clients to determine if value is indeed configured or not.